test/unit/alignment/configuration/align_config_common_test.cpp

```diff
@@ -14,80 +14,82 @@
+    std::pair<cfg::output_sequence1_id, seqan3::type_list<cfg::output_sequence1_id>>,
+    std::pair<cfg::output_sequence2_id, seqan3::type_list<cfg::output_sequence2_id>>,
+    std::pair<cfg::output_score, seqan3::type_list<cfg::output_score>>,
+    std::pair<cfg::output_begin_position, seqan3::type_list<cfg::output_begin_position>>,
+    std::pair<cfg::output_end_position, seqan3::type_list<cfg::output_end_position>>,
+    std::pair<cfg::output_alignment, seqan3::type_list<cfg::output_alignment>>,
+    // other configs
+    std::pair<cfg::band_fixed_size, seqan3::type_list<cfg::band_fixed_size>>,
+    std::pair<cfg::detail::debug, seqan3::type_list<cfg::detail::debug>>,
+    std::pair<cfg::gap_cost_affine, seqan3::type_list<cfg::gap_cost_affine>>,
+    std::pair<cfg::min_score, seqan3::type_list<cfg::min_score, cfg::method_local>>,
+    std::pair<cfg::on_result<callback_t>, seqan3::type_list<cfg::on_result<callback_t>>>,
+    std::pair<cfg::parallel, seqan3::type_list<cfg::parallel>>,
+    std::pair<cfg::detail::result_type<alignment_result_t>, seqan3::type_list<cfg::detail::result_type<alignment_result_t>>>,
+    std::pair<cfg::score_type<int32_t>, seqan3::type_list<cfg::score_type<int32_t>>>,
+    std::pair<cfg::scoring_scheme<nt_scheme>, seqan3::type_list<cfg::scoring_scheme<nt_scheme>>>,
+    std::pair<cfg::vectorised, seqan3::type_list<cfg::vectorised>>
+    >;
 
-TYPED_TEST(alignment_configuration_test, configuration_exists)
-{
-    seqan3::configuration cfg{TypeParam{}};
-    EXPECT_TRUE(decltype(cfg)::template exists<TypeParam>());
-}
+// The pure list of configuration elements to instantiate the typed test case with.
+using align_config_types = pure_config_type_list<align_config_and_taboo_types>;
 
-template <typename t, typename cfg_t>
-void helper_exists()
+template <typename config_t>
+class test_fixture
 {
-    EXPECT_TRUE(cfg_t::template exists<t>());
-}
+public:
+    // The actual config type that is tested.
+    using config_type = config_t;
+    // The taboo list associated with the given TypeParam element.
+    using taboo_list_type =
+        typename seqan3::list_traits::at<seqan3::list_traits::find<config_type, align_config_types>, // determine the index
+                                         align_config_and_taboo_types>::second_type; // extract the taboo list.
 
-template <template <typename ...> typename t, typename cfg_t>
-void helper_exists()
-{
-    EXPECT_TRUE(cfg_t::template exists<t>());
-}
+    // A compatible configuration type for the current configuration element to test.
+    using compatible_configuration_type = make_pipeable_configuration<align_config_types, taboo_list_type>;
 
-TYPED_TEST(alignment_configuration_test, configuration_exists_template)
-{
-    seqan3::configuration cfg{TypeParam{}};
-    helper_exists<TypeParam, decltype(cfg)>();
-}
+    using config_id_type = seqan3::detail::align_config_id;
+
+    static constexpr int8_t config_count = seqan3::detail::align_config_id::SIZE;
```

Comment:
I think this should worK?
(same for search config)

Reply:
I think this has a static value to double check that the config size is correct.

If we now set that value like this, I think you won't be aware that you missed to add something.